platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/AbstractProfileCommand.java

```diff
@@ -0,0 +1,229 @@
+        }
+        if (featuresService == null) {
+            throw new IllegalStateException("FeaturesService not found");
+        }
+        if (bundleService == null) {
+            throw new IllegalStateException("BundleService not found");
+        }
+        try {
+            doExecute(applicationService, featuresService, bundleService);
+        } catch (ApplicationServiceException ase) {
+            printError("Encountered error while trying to perform command. Check log for more details.");
+            LOGGER.debug("Error while performing command.", ase);
+        }
+        return null;
+    }
+
+    /**
+     * Execute profile interaction
+     * be sent to the console and nothing should be returned.
+     *
+     * @param applicationService
+     * @param featuresService
+     * @param bundleService
+     */
+    protected abstract void doExecute(ApplicationService applicationService,
+                            FeaturesService featuresService,
+                            BundleService bundleService)
+            throws Exception;
+
+
+    /**
+     * Get the {@link Feature} that makes up a profile
+     * @param applicationService {@link ApplicationService}
+     * @param profileName Name of profile to get
+     * @return {@link Feature}
+     */
+    protected Optional<Feature> getProfile(ApplicationService applicationService, String profileName) {
+        return Optional.ofNullable(applicationService.getInstallationProfiles().stream()
+                .filter(i -> i.getName().replaceFirst("profile-", "").equals(profileName))
+                .findFirst().orElse(null));
+    }
+
+    /**
+     * Gets the content of an advanced profile
+     * @param profileName Name of profile to get
+     * @return {@link Map}
+     */
+    protected Optional<Map<String, List<String>>> getProfile(String profileName) {
+        File profileFile = Paths.get(profilePath.toString(), profileName + ".json").toFile();
+        Map<String, List<String>> profileMap = null;
+
+        try {
+            profileMap = (Map<String, List<String>>) fromJson(FileUtils.readFileToString(profileFile, StandardCharsets.UTF_8));
+        } catch (FileNotFoundException e) {
+            LOGGER.debug("The file associated with profile: {} was not found under {}",
+                    profileName, profilePath);
+        } catch (IOException e) {
+            LOGGER.debug("An IOException occurred: ", e);
+        }
+
+        return Optional.ofNullable(profileMap);
+    }
+
+    /**
+     * Console output styling for a section heading
+     * @param heading Section Name
+     */
+    public void printSectionHeading(String heading) {
+        console.print(ansi()
+                .fgBrightDefault().bold().a(heading)
+                .newline()
+                .reset());
+    }
+
+    /**
+     * Console output styling for a successful operation for a given item
+     * formatted as <code>Operation Message: item</code>
```

Comment:
✏️  Should this be `message item` instead of `Operation Message: item` to match both argument names and what the code already does? Should the code add the `:` for the caller?
Same comment for all the other print methods.